app/models/history.rb

```diff
@@ -58,6 +58,11 @@ class History < ApplicationRecord
   def self.errored_report_reminder_group_of_patients(patients: nil, created_by: 'Sara Alert System', comment: 'Failed Contact Attempt', error_message: nil)
     histories = []
     patients.uniq.each do |pat|
+      # Avoid adding multiple history items if HoH has errored contact attempts for
+      # contact attempts to them on behalf of their dependents. This would be the
+      # case for failed weblinks
+      next if !pat&.histories&.last.nil? && ((Time.now - pat&.histories&.last&.created_at) / 60) < 5
```

Comment:
This PR is targeting a branch we haven't reviewed yet, so I'm not super clear on how `errored_report_reminder_group_of_patients` is going to fit in yet. Could you explain the logic behind this formula specifically?

Reply:
If a SMS-Weblink HoH with dependents has a phone number that results in errored SMS sends they will end up getting a failed history item for themselves and each of the dependents. This checks to see if they had a history item within the last 5 minutes to avoid a pile up of all the same history items each day during the failed SMS sends. I just realized though that I should add a check to see if the last history item was created within 5 minutes _and_ the history item is an errored contact attempt type. -------- The 5 minute threshold is totally up for adjustment